document_page_reference/models/document_page.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import ValidationError
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    from jinja2.sandbox import SandboxedEnvironment
+    from jinja2 import Undefined
+
+    class Context(SandboxedEnvironment.context_class):
+        def resolve(self, key):
+            res = super().resolve(key)
+            if not isinstance(res, Undefined):
+                return res
+            return self.parent['ref'](key)
+
+    class Environment(SandboxedEnvironment):
+        context_class = Context
+
+    mako_template_env = Environment(
+        block_start_string="<%",
+        block_end_string="%>",
+        variable_start_string="${",
+        variable_end_string="}",
+        comment_start_string="<%doc>",
+        comment_end_string="</%doc>",
+        line_statement_prefix="%",
+        line_comment_prefix="##",
+        trim_blocks=True,               # do not output newline after blocks
+        autoescape=False,
+    )
+except Exception:
+    _logger.error("Jinja2 is not available")
+
+
+class DocumentPage(models.Model):
+
+    _inherit = 'document.page'
+
+    reference = fields.Char()
+    content_parsed = fields.Html(compute='_compute_content_parsed')
+
+    @api.depends('history_head')
+    def _compute_content_parsed(self):
+        for record in self:
+            record.content_parsed = record.get_content()
+
+    @api.constrains('reference')
```

Comment:
It is easier to write it like this
    _sql_constraints = [
        ('reference_uniq', 'unique (reference)', "Reference must be unique !"),
    ]

Reply:
Well, reference could be not defined, and also, we must check the format, so I think it is better to maintain the current constrain